libs/core/src/framework/types.ts

```diff
@@ -118,7 +118,7 @@ export type CommandInput = ZodObject<{
  */
 export type CommandContext<Input extends CommandInput> = {
   readonly actor: Actor;
-  readonly payload: z.infer<Input>;
+  readonly payload: z.input<Input>;
```

Comment:
Using the `input` schema here instead of the `output` schema (equivalent to `infer`) solves the typing issue found when passing partial payloads that get transformed by default zod rules (like in the group.lifecycle.spec below), but it creates the need to use non-null assertions (!) in the rest of query/command payloads (for example in SearchComment.query). 

Probably better to use the "transformed" output schema by definition and hardcode the required default values in test files... we are looking at a simple default transformation here, but it can get more complex than this and will leak into the models.

Reply:
sounds good. I will update.